### ISSUE TITLE
Not learning batchnorm weights during training

### ISSUE THREAD

**Issue by itzsid** [filipradenovic/cnnimageretrieval-pytorch#19]
Hi @filipradenovic,

I am wondering what is the reason behind not learning batch norm weights during training. 

```    # switch to train mode
    model.train()
    model.apply(set_batchnorm_eval)
```
```def set_batchnorm_eval(m):
    classname = m.__class__.__name__
    if classname.find('BatchNorm') != -1:
        # freeze running mean and std
        m.eval()
        # freeze parameters
        # for p in m.parameters():
            # p.requires_grad = False
```

Thanks!

**Comment by itzsid** [comment]
Thanks for the clarification @filipradenovic!
